packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenController.kt

```diff
@@ -0,0 +1,130 @@
+package expo.modules.splashscreen
+
+import android.app.Activity
+import android.os.Handler
+import android.view.View
+import android.view.ViewGroup
+import expo.modules.splashscreen.exceptions.NoContentViewException
+import java.lang.ref.WeakReference
+
+const val SEARCH_FOR_ROOT_VIEW_INTERVAL = 20L
+
+class SplashScreenController(
+    activity: Activity,
+    resizeMode: SplashScreenImageResizeMode,
+    private val rootViewClass: Class<*>,
```

Comment:
I don't really care about this in my code, because I treat this class as a `leaf` of view hierarchy and never enter it's children 😉
But as ReactNative is definitely fetching us with ViewGroup I can go your way 😉